<launch>
   
  <arg name="gui" default="True" />
  <arg name="autostart" default="true"/>
  <arg name="serial_device" default="/dev/ttyUSB0"/>

 
 
  <param name="robot_description" command="$(find xacro)/xacro.py $(find s5fh_controller)/description/urdf/s5fh-standalone.urdf.xacro" />
  <param name="use_gui" value="$(arg gui)"/>
  <!--node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher"-->

  <node name="s5fh_controller" pkg="s5fh_controller" type="s5fh_controller" output="screen">
    <param name="serial_device" value="$(arg serial_device)"/>
    <param name="autostart" value="$(arg autostart)" />
        
    <param name="disable_flags0" value="false" /> <!-- THUMB_FLEXION -->
    <param name="disable_flags1" value="false" /> <!-- THUMB_OPPOSITION // wrist -->
    <param name="disable_flags2" value="false" /> <!-- INDEX_FINGER_DISTAL -->
    <param name="disable_flags3" value="false" /> <!-- INDEX_FINGER_PROXIMAL -->
    <param name="disable_flags4" value="false" /> <!-- MIDDLE_FINGER_DISTAL -->
    <param name="disable_flags5" value="false" /> <!-- MIDDLE_FINGER_PROXIMAL -->
    <param name="disable_flags6" value="false" /> <!-- RING_FINGER -->
    <param name="disable_flags7" value="false" /> <!-- PINKY -->
    <param name="disable_flags8" value="false" /> <!-- FINGER_SPREAD -->
    
    <remap from="s5fh_controller/channel_targets" to="joint_states"/>
  </node>

  <node if="$(arg gui)" name="rqt_gui" pkg="rqt_gui" type="rqt_gui" />

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher">
    <remap from="joint_states" to="s5fh_controller/channel_feedback"/>
  </node>
  
</launch>
